### PR TITLE
Setup spell check and fix issues in markdown, ts and changelogs

### PR DESCRIPTION
**Pull request by timotheeguerin**
- Add spellchecking to CI
- Add `Code Spell Check` vscode extension to list of recommendation
- Fix spell checking errors

### CODE REVIEW

File: common/config/rush/command-line.json [hunk 1]
```diff
@@ -54,6 +54,12 @@
       "summary": "(CUSTOM) Regenerate samples.",
       "shellCommand": "node eng/scripts/npm-run.js regen-samples"
     },
+    {
+      "commandKind": "global",
+      "name": "cspell",
+      "summary": "(CUSTOM) Regenerate samples.",
```
Comment (nguerrera) [inline common/config/rush/command-line.json:60]:
```suggestion
      "summary": "(CUSTOM) Check spelling.",
```

File: cspell.yaml [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+  - proto
+  - protobuf
+  - protoc
+  - regen
+  - rpaas
+  - rushx
+  - safeint
+  - strs
+  - TSES
+  - Uhoh
```
Comment (nguerrera) [inline cspell.yaml:46]: Where do we have this one? LOL.

File: cspell.yaml [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+  - unassignable
+  - Uncapitalize
+  - uninstantiated
+  - unioned
+  - unprojected
+  - unsourced
+  - unversioned
+  - vsix
+  - vswhere
+  - Vtbzpw
```
Comment (nguerrera) [inline cspell.yaml:56]: ?

File: cspell.yaml [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+  - westus
+  - xplat
+  - 𐌰𐌲
+  - 𐌰𐌲𐌰𐌲𐌰𐌲
```
Comment (nguerrera) [inline cspell.yaml:60]: Maybe cspell disable next line for these test cases

File: cspell.yaml (hunk 2 above)
Comment (timotheeguerin) [reply cspell.yaml:46]: image

could rename, just found that one funny to keep :)

File: cspell.yaml (hunk 3 above)
Comment (timotheeguerin) [reply cspell.yaml:56]: its for those samples  Authorization: Basic ZGVtbzpwQDU1dzByZA==.

There was 3 locations for that. Which felt having a bit too much inline suppression

Comment (timotheeguerin) [reply cspell.yaml:56]: But maybe I can have another ignore pattern `/.*==/g`

File: cspell.yaml (hunk 4 above)
Comment (timotheeguerin) [reply cspell.yaml:60]: was a lot of those. But I guess I can do a file exclusion like mike did in his pr
image

Comment (timotheeguerin) [reply cspell.yaml:60]: updated to use file ignore

**Review by bterlson — APPROVED** [review]
I didn't examine all the changes in detail but this seems good regardless.

**Pull request merged**